main/webapp/modules/core/scripts/index/default-importing-controller/controller.js

```diff
@@ -265,10 +265,13 @@ Refine.DefaultImportingController.prototype.getPreviewData = function(callback,
           "limit" : numRows || 100 // More than we parse for preview anyway
         }),
         null,
-        function(data) {
-          result.rowModel = data;
-          callback(result);
-        },
+		  function(data) {
+			  if (data["rows"].length == 0) {
+				alert($.i18n('core-index-import/load-xml-rows-error'));
+			  }
+			  result.rowModel = data;
+			  callback(result);
+		  },
```

Comment:
You have a spacing problem. Also, if you get there, it's because there's an error. It could be a simple network error… (They are VERY common…) Not sure why you check that, and why you don't report other errors.

In terms of user interface, if it fails there, should we go back?

Regards, Antoine

Reply:
The original code was:
https://github.com/OpenRefine/OpenRefine/blob/8e5e06a4b11843bbfcb5d1db128c08aef46d2b5c/main/webapp/modules/core/scripts/index/default-importing-controller/controller.js#L268-L271

Reply:
The original code was:
https://github.com/OpenRefine/OpenRefine/blob/8e5e06a4b11843bbfcb5d1db128c08aef46d2b5c/main/webapp/modules/core/scripts/index/default-importing-controller/controller.js#L268-L271

Actually there is no error. Because of the row limit, if rows < 6  then no rows are shown and no error is mentioned, hence I added the check to atleast mention an error.

Reply:
On further analysis this check is supposed to be in xml-parser-ui.js. Am a pushing commit for further review.